WalletWasabi/Tor/Socks5/Pool/Identities/RandomIdentity.cs

```diff
@@ -0,0 +1,22 @@
+﻿using WalletWasabi.Crypto.Randomness;
+
+namespace WalletWasabi.Tor.Socks5.Pool.Identities
+{
+	public class RandomIdentity : IIdentity
```

Comment:
Can be record too. 

However, thinking about it, I think we don't actually need `DefaultIdentity` and `IIdentity` because `DefaultIdentity` is just once created `RandomIdentity`.

Reply:
I suggested to make sure this equality comparison works well: 
https://github.com/zkSNACKs/WalletWasabi/blob/cb12716c3e80026326d714c75045401144a5bd55/WalletWasabi/Tor/Socks5/Pool/TorHttpPool.cs#L311

Reply:
It was designed for reference equality. That's why I was hesitant to use records but I think records would work too but it would require slightly more CPU work for comparisons.